web/meetup-facilitator/src/App.jsx

```diff
@@ -1,34 +1,58 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
 import './App.css'
+import Button from './components/Button'
+import { useRef, useState } from 'react'
 
 function App() {
-  const [count, setCount] = useState(0)
+
+  const buttonStyles = 'bg-green-700'
+  const groupName = useRef('')
+  const [shouldValidate, setShouldValidate] = useState(false);
+  const [isValid, setIsValid] = useState(true);
+
+
+  const updateGroupName = (e) => {
+    groupName.current = e.target.value;
+    console.log(groupName.current);
+  }
+
+  const onChange = (e) => {
+    updateGroupName(e);
+
+    if (shouldValidate) {
+      validateInput(e);
+    }
+  }
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    console.log("submitting " + groupName.current);
+    setShouldValidate(true);
+
+    // validate once -> TODO: figure out how show message on first submission with invalid input
+    if (groupName.current === '')
```

Comment:
You should validate the input on submit and then here you can just check `isValid` before calling the API.